specifyweb/frontend/js_src/lib/components/Formatters/Formatter.tsx

```diff
@@ -145,51 +149,37 @@ function Definitions({
     }
   }, [needFormatter, handleChange, formatter.definition.fields]);
 
+  const [showConditionalField, setShowConditionalField] = React.useState<
+    WritableArray<boolean>
+  >(trimmedFields.map(() => false));
+
   return table === undefined ? null : (
     <div className="flex flex-col gap-4 divide-y divide-gray-500 [&>*]:pt-4">
       {trimmedFields.map(({ value, fields }, index) => (
-        <div className="flex flex-col gap-2" key={index}>
-          {hasCondition && (
-            <Label.Block>
-              {resourcesText.conditionFieldValue()}
-              <Input.Text
-                isReadOnly={isReadOnly}
-                value={value ?? ''}
-                onValueChange={(value): void =>
-                  handleChanged(
-                    {
-                      value: value.length === 0 ? undefined : value,
-                      fields,
-                    },
-                    index
-                  )
-                }
-              />
-              <span>
-                {index === 0
-                  ? resourcesText.elseConditionDescription()
-                  : resourcesText.conditionDescription()}
-              </span>
-            </Label.Block>
-          )}
-          <Fields
-            fields={[
-              fields,
-              (fields): void => handleChanged({ value, fields }, index),
-            ]}
-            table={table}
-            onDelete={
-              trimmedFields.length < 2
-                ? undefined
-                : (): void =>
-                    handleChange(removeItem(formatter.definition.fields, index))
-            }
-          />
-        </div>
+        <ConditionalFormatter
+          fields={fields}
+          formatter={formatter}
+          hasCondition={hasCondition}
+          index={index}
+          isCollapsed={showConditionalField[index]}
+          key={index}
+          showConditionalField={[showConditionalField, setShowConditionalField]}
+          table={table}
+          trimmedFieldsLength={trimmedFields.length}
+          value={value}
+          onChange={handleChange}
+          onChanged={handleChanged}
+          onRemoveField={(removedIndex: number): void => {
+            const updatedShowConditionalField =
+              Array.from(showConditionalField);
```

Comment:
I would like to use removeItem() but am facing typescript errors. 
`Argument of type 'RA<boolean>' is not assignable to parameter of type 'SetStateAction<WritableArray<boolean>>'.
  The type 'readonly boolean[]' is 'readonly' and cannot be assigned to the mutable type 'WritableArray<boolean>'.`

Reply:
why are you using WritableArray here?
https://github.com/specify/specify7/pull/4381/files/b4b09659c6fb2b5af4d3718fa53dfe7160c0b04e#diff-0ebb7583f8406c8fc8b3d62b05833856ef677a547d0e626fcffd0691db2f833aR153

that is what is causing the issue